src/test/java/com/basho/riak/client/core/operations/itest/ts/ITestFetchOperation.java

```diff
@@ -0,0 +1,57 @@
+package com.basho.riak.client.core.operations.itest.ts;
+
+import com.basho.riak.client.core.RiakFuture;
+import com.basho.riak.client.core.operations.ts.DeleteOperation;
+import com.basho.riak.client.core.operations.ts.FetchOperation;
+import com.basho.riak.client.core.operations.ts.StoreOperation;
+import com.basho.riak.client.core.query.timeseries.Cell;
+import com.basho.riak.client.core.query.timeseries.QueryResult;
+import com.basho.riak.client.core.query.timeseries.Row;
+import com.basho.riak.client.core.util.BinaryValue;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Time Series Fetch Operation Integration Tests
+ * @author Alex Moore <amoore at basho dot com>
+ * @since 2.0.3
+ */
+
+public class ITestFetchOperation extends ITestTsBase
+{
+    @BeforeClass
+    public static void InsertData() throws ExecutionException, InterruptedException
+    {
+        StoreOperation storeOp = new StoreOperation.Builder(tableNameBV).withRows(rows).build();
```

Comment:
Yeah, I agree.  For our purposes though, rewriting the store data again refreshes it to the initial state at least. The only mutable tests are the store (going from null state to rows), and the delete (removing 1 row from the row set). For the deletes I have a special row that I use for that specific test at least.